### PR TITLE
SUBS 1038

### PR DESCRIPTION
**Pull request by davidrichardson**
Sheet mapper using the subs-api

### CODE REVIEW

File: src/main/java/uk/ac/ebi/subs/sheetmapper/SheetMapperService.java [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+
+    private final RestTemplate restTemplate;
+
+    public void mapSheet(Sheet sheet) {
+        UriTemplate submissionUriTemplate  = new UriTemplate(rootApiUrl + "/submissions/{submissionId}");
+        UriTemplate searchUriTemplate  = new UriTemplate(rootApiUrl + "/{type}/search/by-submissionId-and-alias{?submissionId,alias,projection}");
+        UriTemplate createUriTemplate  = new UriTemplate(rootApiUrl + "/submissions/{submissionId}/contents/{type}");
+
+        String targetType = sheet.getTemplate().getTargetType().toLowerCase();
+        String submissionId = sheet.getSubmission().getId();
```
Comment (ke4) [inline src/main/java/uk/ac/ebi/subs/sheetmapper/SheetMapperService.java]: If the sheet is not null, then would it always have a template or submission? Do we need to validate them?

File: src/main/resources/application.properties [hunk 2]
```diff
@@ -1,2 +1,5 @@
 spring.main.web-environment=false
-spring.data.mongodb.database=submissions
\ No newline at end of file
+spring.data.mongodb.database=submissions
```
Comment (ke4) [inline src/main/resources/application.properties]: I think we do not need the 'spring.data.mongodb.database' properties here. It is defined in the secrets repo.

File: src/main/resources/application.properties [hunk 3]
```diff
@@ -1,2 +1,5 @@
...
+usi.tokenservice.username=${usiCoreAAPUsername}
+usi.tokenservice.password=${usiCoreAAPPassword}
+usi.apiRootUrl=http://localhost:8080/api
```
Comment (ke4) [inline src/main/resources/application.properties]: the api root is also defined in subs-api like this: server.contextPath=/api
Do we need both of them?

Comment (davidrichardson) [reply src/main/resources/application.properties]: They serve different purposes. `server.contextPath` is for the server configuration, `usi.apiRootUrl` is config for the client

File: src/main/resources/application.properties (hunk 2 above)
Comment (davidrichardson) [reply src/main/resources/application.properties]: Agreed

File: src/main/java/uk/ac/ebi/subs/sheetmapper/SheetMapperService.java (hunk 1 above)
Comment (davidrichardson) [reply src/main/java/uk/ac/ebi/subs/sheetmapper/SheetMapperService.java]: I'll add some asserts

**Pull request merged**